federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java

```diff
@@ -590,10 +590,9 @@ public void run(KeycloakSession session) {
                         RealmModel currentRealm = session.realms().getRealm(realmId);
                         session.getContext().setRealm(currentRealm);
 
-                        String username = LDAPUtils.getUsername(ldapUser, ldapFedProvider.getLdapIdentityStore().getConfig());
                         exists.value = true;
                         LDAPUtils.checkUuid(ldapUser, ldapFedProvider.getLdapIdentityStore().getConfig());
-                        UserModel currentUser = session.userLocalStorage().getUserByUsername(currentRealm, username);
+                        UserModel currentUser = session.userLocalStorage().getUserById(currentRealm, ldapUser.getUuid());
```

Comment:
The UUID from LDAP doesn't necessarily maps to the ID in Keycloak. There is LDAP_ID attribute, which is used for UUID of LDAP users, so that one should be used for lookup.